### ISSUE TITLE
AWS profile for credstash lookup

### ISSUE THREAD

**Issue by adisharoon** [ansible/ansible#21536]
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
credstash lookup

##### ANSIBLE VERSION
```
ansible 2.2.1.0
  config file = /Users/alecdisharoon/.ansible.cfg
  configured module search path = Default w/o overrides
```

##### CONFIGURATION
N/A
##### OS / ENVIRONMENT
N/A
##### SUMMARY
It would be great if the credstash plugin would accept an AWS profile, the way many AWS modules do.  The credstash python library supports this, it's simply another keyword argument to pass into getSecret (as 'profile_name ', via 'get_session(**kwargs)').  
It's a pain to have to set environment variables for credstash when all other AWS-related Ansible modules allow the use of a boto profile.  
##### STEPS TO REPRODUCE

```
mycoolvar: "{{ lookup('credstash', 'cred_name', region=region, profile=production) }}"
```
and
```
lappy:~ $ cat .aws/credentials
[production]
aws_access_key_id=KJLHKLJHLKJHKLJHK
aws_secret_access_key=ajkdshfkjashfkljahskldjfhasdkljfhaskljhfakd
```

##### EXPECTED RESULTS
This to be supported

##### ACTUAL RESULTS
Not actually supported.

**Comment by jctanner** [comment]
needs_contributor

**Comment by ryansb** [comment]
Closed by #23077
